R/zchunk_L2072.ag_water_irr_mgmt.R

```diff
@@ -1,67 +1,333 @@
+             "L2072.AgCoef_IrrWaterWdraw_bio_mgmt",
+             "L2072.AgCoef_IrrWaterCons_bio_mgmt"))
   } else if(command == driver.MAKE) {
 
     all_data <- list(...)[[1]]
 
+    BlueIrr_m3kg <- BlueIrr_km3 <- TotIrr_m3kg <- TotIrr_km3 <- GreenRfd_m3kg <- water_sector <- water_type <-
+      Prod_Mt <- AgSupplySector <- AgSupplySubsector <- AgProductionTechnology <- minicam.energy.input <-
+      field.eff <- blue_fract <- conveyance.eff <- WaterPrice <- calPrice <- WaterCost <- nonLandVariableCost <-
+      biomass <- type <- region <- GCAM_region_ID <- year <- value <- GCAM_commodity <- GLU <- GLU_name <-
+      IRR_RFD <- MGMT <- coefficient <- . <- NULL  # silence package check notes
+
     # Load required inputs
-    i <- get_data(all_data, "i")
-
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
+    basin_to_country_mapping <- get_data(all_data, "water/basin_to_country_mapping")
+    L132.ag_an_For_Prices <- get_data(all_data, "L132.ag_an_For_Prices")
+    L161.ag_irrProd_Mt_R_C_Y_GLU <- get_data(all_data, "temp-data-inject/L161.ag_irrProd_Mt_R_C_Y_GLU") %>%
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity, -GLU) %>%  # temporary
+      mutate(year = as.integer(substr(year, 2, 5)))
+    L161.ag_rfdProd_Mt_R_C_Y_GLU <- get_data(all_data, "temp-data-inject/L161.ag_rfdProd_Mt_R_C_Y_GLU") %>%
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity, -GLU) %>%  # temporary
+      mutate(year = as.integer(substr(year, 2, 5)))
+    L165.BlueIrr_m3kg_R_C_GLU <- get_data(all_data, "L165.BlueIrr_m3kg_R_C_GLU")
+    L165.TotIrr_m3kg_R_C_GLU <- get_data(all_data, "L165.TotIrr_m3kg_R_C_GLU")
+    L165.GreenRfd_m3kg_R_C_GLU <- get_data(all_data, "L165.GreenRfd_m3kg_R_C_GLU")
+    L165.ag_IrrEff_R <- get_data(all_data, "L165.ag_IrrEff_R")
+    L2052.AgCost_ag_irr_mgmt <- get_data(all_data, "L2052.AgCost_ag_irr_mgmt")
+    L2052.AgCost_bio_irr_mgmt <- get_data(all_data, "L2052.AgCost_bio_irr_mgmt")
+    A03.sector <- get_data(all_data, "water/A03.sector")
+
+    # Primary crops
+    # Compute irrigation water consumption IO coefficients (km3/Mt crop = m3/kg) by region / irrigated crop / year / GLU / management level
+    L165.BlueIrr_m3kg_R_C_GLU %>%
+      ungroup() %>%
```

Comment:
Is this coming in grouped? if so let's fix it upstream in the chunk that's producing this object.